app/controllers/basespace_controller.rb

```diff
@@ -0,0 +1,115 @@
+class BasespaceController < ApplicationController
+  include HttpHelper
+
+  before_action :admin_required
+
+  def oauth
+    @access_token = nil
+
+    if params[:code] &&
+       ENV["BASESPACE_OAUTH_REDIRECT_URI"] &&
+       ENV["BASESPACE_CLIENT_ID"] &&
+       ENV["BASESPACE_CLIENT_SECRET"]
+      response = HttpHelper.post(
+        "https://api.basespace.illumina.com/v1pre3/oauthv2/token",
+        "code" => params[:code],
+        "redirect_uri" => ENV["BASESPACE_OAUTH_REDIRECT_URI"],
+        "client_id" => ENV["BASESPACE_CLIENT_ID"],
+        "client_secret" => ENV["BASESPACE_CLIENT_SECRET"],
+        "grant_type" => "authorization_code"
+      )
+
+      if response.present?
+        @access_token = response["access_token"]
+      end
+    end
+  end
+
+  def projects
+    access_token = params[:access_token]
+
+    if access_token.nil?
+      render json: {
+        error: "basespace access token required"
+      }
+      return
+    end
+
+    # 1024 is the maximum limit allowed by Basespace.
+    # If users reach this limit, we will need to implement multiple requests to fetch all the projects.
```

Comment:
Multiple questions here:
Do we have to set the limit? Does basespace alerts if there are more projects? Will we pass this information to the user at some point?
(In the future, I think we should probably be a bit more strict and load less projects at a time...)

Reply:
If we don't set the limit, it defaults to 10. Basespace does alert if there are more projects.

I deferred implementing the additional fetch if we detect that there is additional data. I could see it being relevant for samples though. Maybe some users will have more than 1024 samples in a project.